src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala

```diff
@@ -140,40 +141,35 @@ final case class AttributeRegistryApiServiceImpl(
       } yield ()
 
       onComplete(result) {
-        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes200)
+        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes204)
       }
     }
 
-  def addNewAttributes(
-    attributeSeed: Seq[AttributeSeed]
-  )(implicit contexts: Seq[(String, String)]): Future[Set[Attribute]] = {
-
-    case class DeltaAttributes(attributes: Set[Attribute], seeds: Set[AttributeSeed]) {
-      def addAttribute(attr: Attribute): DeltaAttributes = copy(attributes = attributes + attr)
-
-      def addSeed(seed: AttributeSeed): DeltaAttributes = copy(seeds = seeds + seed)
-    }
+  private def addNewAttributes(
+    attributesSeeds: Seq[AttributeSeed]
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
 
     // calculating the delta of attributes
-    def delta(attrs: List[Attribute]): DeltaAttributes =
-      attributeSeed.foldLeft[DeltaAttributes](DeltaAttributes(Set.empty, Set.empty))((delta, seed) =>
+    def delta(attrs: Seq[PersistentAttribute]): Set[AttributeSeed] =
+      attributesSeeds.foldLeft[Set[AttributeSeed]](Set.empty)((delta, seed) =>
         attrs
-          .find(persisted => seed.name.equalsIgnoreCase(persisted.name))
-          .fold(delta.addSeed(seed))(delta.addAttribute)
+          .find(persisted => {
+            seed.origin.contains(persisted.origin.getOrElse("")) && seed.code.contains(persisted.code.getOrElse(""))
+          })
+          .fold(delta + seed)(_ => Set.empty)
```

Comment:
which is the purpose of this fold?

Reply:
> which is the purpose of this fold?

The purpose should be that if the seed isn't defined in the persisted attributes we should add it to the delta of new attributes, otherwise it should not be added.

If you think it's better we thought that we could rewrite that as the following:
```
        if(!attrs.exists(persisted => seed.origin == persisted.origin && seed.code == persisted.code)) 
           attributesDelta + seed
        else Set.empty
```

Let us know if you have other suggestions!

Reply:
You are absolutely right 👍